server/base/src/main/java/org/apache/accumulo/server/master/state/TabletStateStore.java

```diff
@@ -53,7 +53,10 @@
   /**
    * Tablet servers will update the data store with the location when they bring the tablet online
    */
-  void setLocations(Collection<Assignment> assignments) throws DistributedStoreException;
+  // void setLocations(Collection<Assignment> assignments) throws DistributedStoreException;
+
+  void setLocations(Collection<Assignment> assignments, TServerInstance prevLastLoc)
```

Comment:
At the hackathon we discussed the following change.  Curious if you are considering this?
```suggestion
  void setLocations(Assignment assignments, TServerInstance prevLastLoc)
```

Reply:
Also still working on this, Ran into the same issues as mentioned above. Particularly in NullTserver.java and MergeStateIT IIRC